src/script/ptl-tool.py

```diff
@@ -141,9 +152,24 @@ def build_branch(args):
         # So we know that we're not on an old test branch, detach HEAD onto ref:
         base.checkout()
 
-    for pr in args.prs:
-        log.info("Merging PR #{pr}".format(pr=pr))
+    prs = args.prs
+    if args.pr_label:
+        payload = {'labels': args.pr_label, 'sort': 'created', 'direction': 'desc'}
+        labeled_prs = requests.get("https://api.github.com/repos/ceph/ceph/issues", auth=(USER, PASSWORD), params=payload)
```

Comment:
See, created the test branch for an invalid pr-label.
```
$ src/script/ptl-tool.py --base master --merge-branch wip-jcollin-testing --pr-label wip-jcollin-testing2 #or wip-aoeu-testing
Detaching HEAD onto base: master
Will merge PRs: []
Checked out new branch wip-jcollin-testing-20171011
Created tag testing/wip-jcollin-testing-20171011_04
jcollin@smithi042:~/test-ptl-oct11$ git branch
  master
  * wip-jcollin-testing-20171011
```

Please check if the branch exists first and drop it. May be move the existing label validation code to a function ?

Reply:
Fixed!

Reply:
`--pr-label ' '`is bypassing it. I have pressed Ctrl+C and it saved me :-D. 

```
jcollin@smithi042:~/test-ptl-oct12$ src/script/ptl-tool.py --base master --merge-branch wip-jcollin-testing --pr-label ' '
Adding labeled PR #18271 to PR list
Adding labeled PR #18270 to PR list
Adding labeled PR #18269 to PR list
Adding labeled PR #18268 to PR list
Adding labeled PR #18267 to PR list
Adding labeled PR #18266 to PR list
Adding labeled PR #18265 to PR list
Adding labeled PR #18264 to PR list
Adding labeled PR #18263 to PR list
Adding labeled PR #18262 to PR list
Adding labeled PR #18261 to PR list
Adding labeled PR #18260 to PR list
Adding labeled PR #18259 to PR list
Adding labeled PR #18257 to PR list
Adding labeled PR #18256 to PR list
Adding labeled PR #18255 to PR list
Adding labeled PR #18254 to PR list
Adding labeled PR #18253 to PR list
Adding labeled PR #18251 to PR list
Adding labeled PR #18250 to PR list
Adding labeled PR #18249 to PR list
Adding labeled PR #18248 to PR list
Adding labeled PR #18247 to PR list
Adding labeled PR #18246 to PR list
Adding labeled PR #18245 to PR list
Adding labeled PR #18244 to PR list
Adding labeled PR #18243 to PR list
Adding labeled PR #18241 to PR list
Adding labeled PR #18240 to PR list
Adding labeled PR #18239 to PR list
Will merge PRs: [18271, 18270, 18269, 18268, 18267, 18266, 18265, 18264, 18263, 18262, 18261, 18260, 18259, 18257, 18256, 18255, 18254, 18253, 18251, 18250, 18249, 18248, 18247, 18246, 18245, 18244, 18243, 18241, 18240, 18239]
Detaching HEAD onto base: master
Merging PR #18271
Labeled PR #18271 wip-jcollin-testing
Merging PR #18270
^CTraceback (most recent call last):
  File "src/script/ptl-tool.py", line 308, in <module>
    
  File "src/script/ptl-tool.py", line 305, in main

```

Reply:
okay, fixed

Reply:
@batrick Thanks. I will test tomorrow. Need some sleep now. :-)

Reply:
@batrick In the mean time, could you clarify what is `label = '-'` ? It is also an invalid label, right ?

Reply:
1. Check the empty string also.
```
$ src/script/ptl-tool.py --base master --merge-branch wip-jcollin-testing --pr-label ''
Will merge PRs: []
Detaching HEAD onto base: master
Checked out new branch wip-jcollin-testing-20171013
Created tag testing/wip-jcollin-testing-20171013
```

> Adding labeled PR #18284 to PR list

2. Do we need to add and merge an unreviewed / unapproved PR for testing ? Do we need to check the PR status before the add / merge ?

Reply:
> Check the empty string also.
done.

> Do we need to add and merge an unreviewed / unapproved PR for testing ? Do we need to check the PR status before the add / merge ?

No, that should stay the ptl's responsibility to check.